### PR TITLE
Implement ZSink and ZTransducer Unwrap Operators

### LINKED ISSUE #5637: ZIO 2.0 Streams: Add unwrap* functions to sink, stream, transducer (?)
These functions are missing or are encoded weirdly (e.g. ZSink.managed).

### PR DESCRIPTION
**Pull request by adamgfraser**
Resolves #5637.

### CODE REVIEW

**Review by jdegoes — COMMENTED** [review]
These look really good! Can we deprecate the `.managed()` operators, which are a combination of unwrap and flatMap?

**Comment by adamgfraser** [comment]
I think we can reform `managed` to just lift a value produced from a managed effect into sink.

**Comment by adamgfraser** [comment]
Actually not if we want `Sink` to represent a static scope as opposed to a dynamic one.

File: streams-tests/shared/src/test/scala/zio/stream/experimental/ZSinkSpec.scala [hunk 1]
```diff
@@ -155,14 +155,13 @@ object ZSinkSpec extends ZIOBaseSpec {
...
+        suite("unwrapManaged")(
           test("happy path") {
             for {
               closed <- Ref.make[Boolean](false)
               res     = ZManaged.acquireReleaseWith(ZIO.succeed(100))(_ => closed.set(true))
-              sink = ZSink.managed[Any, Any, Any, Any, Int, Nothing, (Long, Boolean)](res)(m =>
-                       ZSink.count.mapZIO(cnt => closed.get.map(cl => (cnt + m, cl)))
-                     )
+              sink =
+                ZSink.unwrapManaged(res.map(m => ZSink.count[Any].mapZIO(cnt => closed.get.map(cl => (cnt + m, cl)))))
```
Comment (jdegoes) [inline streams-tests/shared/src/test/scala/zio/stream/experimental/ZSinkSpec.scala:164]: Good bye type parameters ❤️

**Review by jdegoes — APPROVED** [review]

**Pull request merged**